### PR TITLE
Use pre-commit in project

### PR DESCRIPTION
**Pull request by BeyondEvil**
Since `black` wasn't used previously, a lot of the changes are due to running it.

### CODE REVIEW

File: tests/test_verify_base_url.py [hunk 1]
```diff
@@ -2,70 +2,62 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import pytest
 from requests.packages.urllib3.util.retry import Retry
 
 
-@pytest.fixture(autouse=True)
-def httpserver(httpserver):
-    return httpserver
```
Comment (BeyondEvil) [inline tests/test_verify_base_url.py]: For some reason, some of the tests fails when using `python 2.7` without this, seemingly, redundant fixture. Any idea why, @nicoddemus ?

Tried replacing it with `pytestmark = pytest.mark.usefixture("httpserver")` without success.

Comment (BeyondEvil) [reply tests/test_verify_base_url.py]: Or maybe @davehunt knows/remembers?

Comment (davehunt) [reply tests/test_verify_base_url.py]: I would agree this looks redundant, but regretfully I don't recall why this was necessary.

File: development.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
+Development
+===========
+
+To contribute to `pytest-html` you can use `Pipenv`_ to manage
```
Comment (davehunt) [inline development.rst]: pytest-base-url

File: tests/test_verify_base_url.py (hunk 1 above)
Comment (BeyondEvil) [reply tests/test_verify_base_url.py]: I just added it back, with a note about removing it when we drop 2.7 support.

File: development.rst (hunk 2 above)
Comment (BeyondEvil) [reply development.rst]: Done

**Pull request merged**